packages/roosterjs-content-model-types/lib/event/SnapshotLogicalRootEvent.ts

```diff
@@ -0,0 +1,20 @@
+import type { EntityState } from '../parameter/FormatContentModelContext';
+import type { BasePluginEvent } from './BasePluginEvent';
+
+/**
+ * Undo snapshot event when a custom logical root is used (logicalRoot != physicalRoot)
+ * This event is triggered when an undo snapshot is taken
+ * Plugins can handle this event to include entity state to include in the snapshot
+ */
+export interface SnapshotLogicalRootEvent extends BasePluginEvent<'snapshotLogicalRoot'> {
```

Comment:
Merge this event into EntityOperation event.

Reply:
And use a better name since the event is just to tell plugin we are adding undo snapshot under entity, plugin can generate its own undo snapshot, and can also do other things.